go-update/src/main/java/org/reactome/release/goupdate/DuplicateCleaner.java

```diff
@@ -0,0 +1,183 @@
+
+	public static void main(String[] args)
+	{
+		String pathToResources = "src/main/resources/go-update.properties";
+		if (args.length > 0)
+		{
+			pathToResources = args[0];
+		}
+		try
+		{
+			Properties props = new Properties();
+			props.load(new FileInputStream(pathToResources));
+			DuplicateCleaner step = new DuplicateCleaner();
+			step.executeStep(props);
+		}
+		catch (Exception e)
+		{
+			e.printStackTrace();
+		}
+
+	}
+
+	@Override
+	public void executeStep(Properties props) throws Exception
+	{
+		MySQLAdaptor adaptor = DuplicateCleaner.getMySQLAdaptorFromProperties(props);
+		int totalDuplicatesCount = 0;
+		int instancesWithSignificantReferrers = 0;
+		
+		DuplicateReporter dupeReporter = new DuplicateReporter(adaptor);
+		Set<Long> dbIDsToDelete = new HashSet<Long>();
+		Set<Long> dbIDsToKeep = new HashSet<Long>();
+		Map<String, Integer> duplicates = dupeReporter.getDuplicateAccessions();
+		
+		// loop through all duplicated accessions.
+		for (String accession : duplicates.keySet())
+		{
+			totalDuplicatesCount = duplicates.keySet().size();
+			logger.info("Accession " + accession +" is duplicated " + duplicates.get(accession) + " times.");
+			List<Long> dbIDsWithNoReferrers = new ArrayList<Long>();
+			Map <Long,Integer> refCounts = dupeReporter.getReferrerCountForAccession(accession, ReactomeJavaConstants.GO_BiologicalProcess, ReactomeJavaConstants.GO_MolecularFunction, ReactomeJavaConstants.GO_CellularComponent, ReactomeJavaConstants.Compartment);
+			// if there are referrers...
+			if (refCounts != null && refCounts.size() > 0)
+			{
+				// for each DB ID in referrer counts
+				for (Long dbId : refCounts.keySet())
+				{
+					// if there are referrers.
+					if (refCounts.get(dbId) > 0)
+					{
+						logger.info("\tAccession instance with DB_ID "+dbId + " has "+refCounts.get(dbId) + " significant referrers");
+						// Add to list of IDs to keep (I think this set is no longer needed...)
+						dbIDsToKeep.add(dbId);
+						// update count.
+						instancesWithSignificantReferrers ++;
+						GKInstance inst = adaptor.fetchInstance(dbId);
+						@SuppressWarnings("unchecked")
+						Collection<GKSchemaAttribute> refAttribs = (Collection<GKSchemaAttribute>) inst.getSchemClass().getReferers();
+						// check each atttribute
+						for (GKSchemaAttribute refAtt : refAttribs)
+						{
+							@SuppressWarnings("unchecked")
+							Collection<GKInstance> refs = (Collection<GKInstance>) inst.getReferers(refAtt);
+							if (refs != null && refs.size() > 0)
+							{
+								for (GKInstance ref : refs)
+								{
+									if (!Arrays.asList(ReactomeJavaConstants.GO_BiologicalProcess, ReactomeJavaConstants.GO_MolecularFunction, ReactomeJavaConstants.GO_CellularComponent, ReactomeJavaConstants.Compartment).contains(ref.getSchemClass().getName()))
+									{
+										logger.info("\t\tvia "+refAtt.getName()+":\t"+ref.toString());
+									}
+								}
+							}
+						}
```

Comment:
It took me some time to understand this block -- could it be extracted to a method called `logSignificantReferrers(inst)`?

Reply:
I think it's better where it is, but I'll updated the comments in this are.